db/version_edit.h

```diff
@@ -34,20 +30,22 @@ extern uint64_t PackFileNumberAndPathId(uint64_t number, uint64_t path_id);
 struct FileDescriptor {
   // Table reader in table_reader_handle
   TableReader* table_reader;
-  uint64_t packed_number_and_path_id;
+  uint32_t path_id;
+  uint64_t file_number;
```

Comment:
Do we have to waste this 8 bytes? Can you explain why it is absolutely needed?

Reply:
The motivation for isolating the path_id to its own integer is that: the current way of packing path id and file number only gives a limit of `4` for path id.

I didn't realize that `FileDescriptor` can leave a big memory footprint, so took the simple approach of giving path id its own integer to avoid the casting.

Glad you asked. I made a change to give 2 more bits to path id so that the upper limit now goes up to `16`, which should be sufficient for majority cases. However, it does shrink the upper limit of file numbers quite a bit. It is still a very large upper limit so I'm assuming that it should be fine.

Let me know if this works.